contracts/HATArbitrator.sol

```diff
@@ -0,0 +1,161 @@
+
+    modifier onlyExpertCommittee() {
+        if (msg.sender != expertCommittee) {
+            revert OnlyExpertCommittee();
+        }
+        _;
+    }
+
+    modifier onlyChallengedActiveClaim(bytes32 _claimId) {
+        (bytes32 claimId,,,,,uint32 challengedAt,,,,,,,) = vault.activeClaim();
+
+        if (claimId != _claimId) {
+            revert ClaimDisputedDoesNotExist();
+        }
+
+        if (challengedAt == 0) {
+            revert ClaimIsNotDisputed();
+        }
+        _;
+    }
+
+    modifier onlyUnresolvedDispute(bytes32 _claimId) {
+        if (resolvedAt[_claimId] != 0) {
+            revert AlreadyResolved();
+        }
+        _;
+    }
+
+    modifier onlyResolvedDispute(bytes32 _claimId) {
+        if (resolvedAt[_claimId] == 0) {
+            revert NoResolution();
+        }
+        _;
+    }
+
+    constructor (IHATVault _vault, address _expertCommittee, IERC20 _token, uint256 _disputeStartingAmount, uint256 _minBondAmount) {
+        vault = _vault;
+        expertCommittee = _expertCommittee;
+        token = _token;
+        disputeStartingAmount = _disputeStartingAmount;
+        minBondAmount = _minBondAmount;
+        if (minBondAmount > disputeStartingAmount) {
+            revert DisputeStartingAmountMustBeHigherThanMinAmount();
+        }
+    }
+
+    function dispute(bytes32 _claimId, bytes32 _ipfsHash, uint256 _bondAmount) external {
+        if (_bondAmount < minBondAmount) {
+            revert BondAmountSubmittedTooLow();
+        }
+
+        (bytes32 claimId,,,,,uint32 challengedAt,,,,,,,) = vault.activeClaim();
+        if (claimId != _claimId) {
+            revert ClaimDisputedDoesNotExist();
+        }
+
+        disputersBonds[msg.sender][_claimId] += _bondAmount;
+        totalBondsOnClaim[_claimId] += _bondAmount;
+
+        token.safeTransferFrom(msg.sender, address(this), _bondAmount);
+
+        if (totalBondsOnClaim[_claimId] >= disputeStartingAmount) {
+            if (challengedAt == 0) {
+                vault.challengeClaim(_claimId);
+            } else {
+                // solhint-disable-next-line not-rely-on-time
+                if (block.timestamp > challengedAt + 24 hours) {
+                    revert CannotSubmitMoreEvidence();
+                }
+            }
+        }
+
+        emit ClaimDisputed(_claimId, msg.sender, _ipfsHash, _bondAmount);
+    }
+
+    function dismissDispute(bytes32 _claimId) external onlyExpertCommittee onlyChallengedActiveClaim(_claimId) onlyUnresolvedDispute(_claimId) {
+        token.safeTransferFrom(msg.sender, address(this), totalBondsOnClaim[_claimId]);
+
+        vault.approveClaim(_claimId, 0, address(0));
+    }
+
+    function acceptDispute(bytes32 _claimId, Resolution calldata _resolution) external onlyExpertCommittee onlyChallengedActiveClaim(_claimId) onlyUnresolvedDispute(_claimId) {
+        resolutions[_claimId] = _resolution;
+        resolvedAt[_claimId] = block.timestamp;
```

Comment:
why keep this separatley, why not have a `resolvedAt` as part of the `Resolution` struct?